src/Mollie/WC/Plugin.php

```diff
@@ -694,6 +725,44 @@ public static function addGateways( array $gateways ) {
 		return $gateways;
 	}
 
+    /**
+     * Disable Bank Transfer Gateway
+     *
+     * @param array $gateways
+     * @return array
+     */
+    public static function maybeDisableBankTransferGateway(array $gateways)
+    {
+        $isWcApiRequest = (bool)filter_input(INPUT_GET, 'wc-api', FILTER_SANITIZE_STRING);
+        $bankTransferSettings = get_option('mollie_wc_gateway_banktransfer_settings', false);
+        $isSettingActivated = false;
+        if($bankTransferSettings){
+            $expiryDays = $bankTransferSettings['activate_expiry_days_setting'];
+            $isSettingActivated = mollieWooCommerceStringToBoolOption($expiryDays);
+        }
+
+        /*
+         * There is only one case where we want to filter the gateway and it's when the
+         * pay-page render the available payments methods AND the setting is enabled
+         *
+         * For any other case we want to be sure bank transfer gateway is included.
+         */
+        if ($isWcApiRequest ||
+            !$isSettingActivated ||
+            is_checkout() && ! is_wc_endpoint_url( 'order-pay' )||
+            !wp_doing_ajax() && ! is_wc_endpoint_url( 'order-pay' )||
+            is_admin()
+        ) {
+            return $gateways;
+        }
+        $bankTransferGatewayClassName = 'Mollie_WC_Gateway_BankTransfer';
```

Comment:
I guess, we can make it more flexible by using `Mollie_WC_Gateway_BankTransfer::class`. This way, we can expect this code will keep working if that class will be renamed. Or, you would be able to find this place by clicking on the class name in your IDE.